### PR TITLE
Give Tab ownership of its SwitchToTab command

### PR DESCRIPTION
**Pull request by leonMSFT**
Currently, `CommandPalette` creates and maintains the `SwitchToTab`
commands used for the ATS. When `Command` goes into the
TerminalSettingsModel, the palette won't be able to access `Command`'s
implementation type, making it difficult for `CommandPalette` to tell
`Command` to listen to `Tab` for changes.

This PR changes the relationship up so `Tab` now manages its
`SwitchToTab` command, and `CommandPalette` just plops the command from
`Tab` into its list.

### CODE REVIEW

File: src/cascadia/TerminalApp/Tab.idl [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+import "Command.idl";
+
 namespace TerminalApp
 {
     [default_interface] runtimeclass Tab : Windows.UI.Xaml.Data.INotifyPropertyChanged
     {
         String Title { get; };
         Windows.UI.Xaml.Controls.IconSource IconSource { get; };
+        Command SwitchToTabCommand { get; };
+        UInt32 TabViewIndex { get; };
```
Comment (DHowett) [inline src/cascadia/TerminalApp/Tab.idl:13]: feel free to just call this "Index"

**Review by DHowett — APPROVED** [review]
I'm cool with this. I prefer it over the design where the command listens to the tab, for sure.

Eventually, this may help us disentangle the "command"-based nature of switching? :smile:

**Review by carlos-zamora — COMMENTED** [review]
You're amazing. Thanks!

**Review by carlos-zamora — APPROVED** [review]
Apparently Ctrl+Enter does _not_ approve...

**Pull request merged**